File: ios/MyClipboard/ClipboardX.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ClipboardX, NSObject)
RCT_EXTERN_METHOD(setString:(NSString *)text
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getString:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
+ (BOOL)requiresMainQueueSetup;
@end

File: ios/MyClipboard/ClipboardX.swift
import Foundation
import UIKit

@objc(ClipboardX)
class ClipboardX: NSObject {
  @objc(setString:resolver:rejecter:)
  func setString(_ text: NSString,
                 resolver resolve: RCTPromiseResolveBlock,
                 rejecter reject: RCTPromiseRejectBlock) {
    UIPasteboard.general.string = text as String
    resolve(nil)
  }

  @objc(getString:rejecter:)
  func getString(_ resolve: RCTPromiseResolveBlock,
                 rejecter reject: RCTPromiseRejectBlock) {
    resolve(UIPasteboard.general.string ?? "")
  }

  @objc static func requiresMainQueueSetup() -> Bool { true }
}
